src/main/java/graphql/execution/instrumentation/dataloader/PerLevelDataLoaderDispatchStrategy.java

```diff
@@ -0,0 +1,235 @@
+package graphql.execution.instrumentation.dataloader;
+
+import graphql.Assert;
+import graphql.Internal;
+import graphql.execution.DataLoaderDispatchStrategy;
+import graphql.execution.ExecutionContext;
+import graphql.execution.ExecutionStrategyParameters;
+import graphql.execution.FieldValueInfo;
+import graphql.execution.MergedField;
+import graphql.schema.DataFetcher;
+import org.dataloader.DataLoaderRegistry;
+
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+
+@Internal
+public class PerLevelDataLoaderDispatchStrategy implements DataLoaderDispatchStrategy {
```

Comment:
can we unit test this  eg make a series of calls to it as if it was a real engine??